### DISCUSSION TITLE
Including task name instead of task template in pod name

### DISCUSSION THREAD

**Discussion by andyshieh** [argoproj/argo-workflows#7631] [Q&A]
Dear Argo community, is there currently an option to have the Argo generated pod names include the actual task/step name instead of the task template name?
So something like [workflow id]-[task name]-[random numeric string] instead of [workflow id]-[task template]-[random numeric string]

The reason for this is that very often we would run several similar tasks based on one single template, and we would use the actual task name to indicate the parameter. Using the task name instead of the template name makes it even easier to identify the right pods for troubleshooting.

For example - I might have a task template "process" which is referenced in two tasks "process-a" and "process-b", which these two tasks only differing in a few input parameters. Currently we can't directly tell which pod corresponds to process-a and which corresponds to process-b. 

Thank you for your help!

**Comment by terrytangyuan** [comment]
Use `POD_NAMES` environment variable: 

> Whether to have pod names contain the template name (v2) or be the node id (v1).

See https://github.com/argoproj/argo-workflows/blob/master/docs/environment-variables.md for details.

**Reply by andyshieh** [reply]
Thank you @terrytangyuan . But I'm looking for ways to use the "task name" as the pod name and not the node id.
Sometimes we define and single task template that are used in multiple tasks each with a slightly different task name, with each task differing in terms of certain input parameters.
For example I might define a task template called "process", and have two tasks named as "process-a" and "process-b" both referencing the same "process" template. process-a and process-b would differ in the value of their input parameter values.
POD_NAMES=v1 would not include any task template/name related information in the pod name.
POD_NAMES=v2 would include the task template (in this case, "process") in the pod name, but not the task name, and thus from the pod name I can't directly tell if a pod is running "process-a" or "process-b" until I look into the logs.

Thanks.

**Reply by terrytangyuan** [reply]
This might be what you are looking for: https://github.com/argoproj/argo-workflows/issues/2871

Would the display name be useful in your case?

**Reply by ghost** [reply]
We would like it for the actual pod names in kubernetes, not just in the workflow ui.

**Comment by ridhazorgui** [comment]
Hey, This would be very usefull, Any updates ?

**Comment by droundy** [comment]
Is there any progress on this? Or other approaches?

I've got a helper `WorkflowTemplate` that is generating a huge number of pods (for a big dag) all of which are identically named, which is quite a nuisance.
